JaCoCo based code coverage analysis

rsvoboda opened the pull request:
JaCoCo based code coverage analysis

To get aggregated report I had to go via maven-antrun-plugin. I saw similar approach in other projects too.

Procedure example:
```
mvn -Pjacoco clean verify
mvn -Pjacoco-report verify -DskipTests -Denforcer.skip=true -Dcheckstyle.skip=true -Dskip.gradle.build=true
open target/jacoco-report/index.html

```

Report example: [jacoco-report.zip](https://github.com/quarkusio/quarkus/files/3018884/jacoco-report.zip)

rsvoboda committed:
JaCoCo based code coverage analysis

cescoffier commented on pom.xml:
```diff
@@ -170,6 +171,125 @@
+                                            <structure name="JaCoCo for Quarkus">
+                                                <classfiles>
+                                                    <fileset dir="${basedir}">
+                                                        <include name="**/*999-SNAPSHOT.jar"/>
```
Can you use ${project.version} instead?

cescoffier commented on pom.xml:
```diff
@@ -170,6 +171,125 @@
+                                                <classfiles>
+                                                    <fileset dir="${basedir}">
+                                                        <include name="**/*999-SNAPSHOT.jar"/>
+                                                        <exclude name="**/*quarkus-cli-999-SNAPSHOT.jar"/>
```
Any reason to exclude this one?

rsvoboda replied:
JaCoCo doesn't like duplicated classes in and it throws 
`[ERROR] around Ant part ...<report>... @ 8:11 in /Users/rsvoboda/git/quarkus/target/antrun/build-main.xml: Error while analyzing devtools/aesh/target/original-quarkus-cli-999-SNAPSHOT.jar@META-INF/versions/11/org/wildfly/common/archive/JDKSpecific.class. Can't add different class with same name: org/wildfly/common/archive/JDKSpecific`

cescoffier commented on pom.xml:
```diff
@@ -170,6 +171,125 @@
             </build>
         </profile>
+        <profile>
+            <id>jacoco</id>
```
Can you add as a comment the instruction to run these?

rsvoboda commented:
@cescoffier PR updated

rsvoboda committed:
JaCoCo - prevent instrumentation of test classes, more generic report excludes

rsvoboda commented:
Rebased and added commit to this PR to:

 - have faster report generation command
 - avoid instrumenting test classes (io.quarkus.it.*)
 - have more generic report excludes

Sanne commented:
@cescoffier your "requests for changes" is blocking this PR. I guess it's out of date?
